### ISSUE TITLE
Need to update content://URI for compatibility with google

### ISSUE THREAD

**Issue by techguyone** [openintents/notepad#38]
currently only reads content:// URI generated within their app, not from other apps. 

Play Store will require this starting November 2018.

**Comment by friedger** [comment]
Could you please provide a link with more information?

tgo <notifications@github.com> schrieb am Di., 3. Apr. 2018, 12:14:

**Comment by techguyone** [comment]
Looking at the OI Note source code here: https://github.com/openintents/notep...oteEditor.java

It looks that code is intended to read a content:// URI generated within their app, not from other apps. They're behaving as though any content:// URI sent a reference to a note in their database with a bunch of proprietary columns FX of course isn't providing these. 

OI Note does work fine if you have FX use file:// URIs to open files, at was the default in FX 6.3 and prior. This setting can be changed in FX's settings, "Opening Files", "Use Content URIs" (last option). This option will be removed in a not-too-distant version of FX, as the Play Store will require this starting November 2018.

TL/DR you can't open OI Note in a different app for example using a file explorer and using 'open with'  because OI isn't set up that way, if it isn't changed to do so, in November it won't work.

**Comment by friedger** [comment]
Thanks for the clarification.

So, you would like to read a text file from FX, or any other openable
stream that comes from FX. Correct?

Should OI Notepad import that file into a note? Or keep it as external and
delegate and updates back to the app the content uri came from? If that is
possible at all.

Maybe OI notepad should store the external uri and ask the user to update
the imported note when the user opens it a second time. Writing back to the
external uri seems difficult.

tgo <notifications@github.com> schrieb am Do., 5. Apr. 2018, 16:29:

**Comment by techguyone** [comment]
I liked the simplicity we had before, I just navigated to  my SD Card using FX Explorer (would be the same for any file manager) and long selecting the txt file so you get the 'open with' dialog come up, then choosing 'open with OI Notes' in this case

This issue only came to light as the dev of FX Explorer has changed his method of opening in preparation of Google guidelines later in the year otherwise I wouldn't have known there was a future problem.

At least now you can pre empt the change and OI Notes will still work when opening via another app.

As far as the mechanics go, I'm guessing that once OI opens the tex, its then editable and savable if required the same as if we'd had OI as the default txt viewer and opened it directly, I don't see the need to make it complicated.

**Comment by aperomsik** [comment]
Still not seeing where you provided a link to the policy change. Did I miss it? We need to understand how it will affect the use case of notes embedded in OI Shopping List items.

**Comment by friedger** [comment]
Aaron, file uris are not allowed to be handed over to other apps anymore.

Aaron Peromsik <notifications@github.com> schrieb am Do., 5. Apr. 2018,
18:51:

**Comment by techguyone** [comment]
File URI's no, but Content URI's yes (I have no idea what the difference is , but the Dev of FX Explorer seemed to know what it meant)

**Comment by techguyone** [comment]
I'll just add this for the necessary proof

Here is the requirement regarding Play Store API level requirements:

https://developer.android.com/distri...arget-sdk.html

It requires new apps to Target SDK 26 (Oreo 8.0) by August 2018 and existing apps to do by November 2018.

Starting in SDK 24 (Nougat 7.0) apps are forbidden from sending files to each other using a file:/// URI and must instead use a content:// URI. This is mentioned in the Play Store info URL, but also more specifically here: https://developer.android.com/about/...#sharing-files
